docs/java-rest/high-level/document/exists.asciidoc

```diff
@@ -0,0 +1,59 @@
+[[java-rest-high-document-exists]]
+=== Exists API
+
+[[java-rest-high-document-exists-request]]
+==== Exists Request
+
+The `exists()` API is `true` if a document exists, and `false` otherwise.
+It uses `GetRequest` just like the <<java-rest-high-document-get>>.
+All of its <<java-rest-high-document-get-request-optional-arguments, optional arguments>>
+are supported. Since `exists()` only returns `true` or false, we recommend
+turning off fetching `_source` and any stored fields so the request is
+slightly lighter:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[exists-request]
+--------------------------------------------------
+<1> Index
+<2> Type
+<3> Document id
+<4> Disable fetching `_source`.
+<5> Disable fetching stored fields.
+
+[[java-rest-high-document-exists-sync]]
+==== Synchronous Execution
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/CRUDDocumentationIT.java[exists-execute]
+--------------------------------------------------
+
+[[java-rest-high-document-exists-async]]
+==== Asynchronous Execution
+
+The asynchronous execution of `exists()` requires both the `GetRequest`
```

Comment:
In most docs this section starts something like : 
[Update API](https://www.elastic.co/guide/en/elasticsearch/client/java-rest/master/java-rest-high-document-update.html#java-rest-high-document-update-async)
```
The asynchronous execution of an update request requires both the UpdateRequest instance and an ActionListener instance to be passed to the asynchronous method:
```
Would it be more consistent if : 
s/`exists()`/exists request/

Reply:
I can do that. I'm not sure it is better, but it is more consistent.